nemo_skills/inference/generate.py

```diff
@@ -271,8 +272,14 @@ def __init__(self, cfg: GenerateSolutionsConfig):
         if self.cfg.use_completions_api or self.cfg.server.get("enable_soft_fail", False):
             # These are the only cases where we need a tokenizer
             self.tokenizer = self.cfg.tokenizer or self.cfg.server["model"]
+            try:
```

Comment:
then we want to do it outside of this if, right? As we always want to try to use it for prompts?

on a second thought, how about we maybe don't log prompt length by default as for some cases (e.g. ruler) this will incur a non-trivial cost and might slow down generations. So maybe we add a parameter `count_prompt_tokens` that's false by default, but can be set to enable this? And then if it's set, we try to get tokenizer, but if we can't we actually hard-fail as users explicitly asked for that, but we can't do it